### PR TITLE
Update bats to support no Pulp 2

### PR DESCRIPTION
**Pull request by ehelms**
This is also part of EL8 support which does not have Pulp 2.

### CODE REVIEW

File: bats/fb-content-katello.bats [hunk 1]
```diff
@@ -211,32 +215,44 @@ setup() {
 }
 
 @test "install katello-agent" {
+  tSkipIfPulp2
```
Comment (ekohl) [inline bats/fb-content-katello.bats:218]: The naming of this feels a bit odd. This reads like it should skip katello-agent tests if it's Pulp 2 but the implementation suggests it's skipping unless Pulp 2 is present

**Comment by ehelms** [comment]
Hah, I forgot the 'No' in there

File: bats/foreman_helper.bash [hunk 2]
```diff
@@ -17,3 +17,13 @@ tForemanVersion() {
     fi
   ) | cut -d. -f1-2
 }
+
+Pulp2() {
```
Comment (evgeni) [inline bats/foreman_helper.bash:21]:
```suggestion
tIsPulp2() {
```

File: bats/foreman_helper.bash [hunk 3]
```diff
@@ -17,3 +17,13 @@ tForemanVersion() {
...
+  tPackageExists pulp-server
+}
+
+tSkipIfNoPulp2() {
+  if ! tIsPulp2; then
+   skip "Puppet content not available with Pulp 3 only"
```
Comment (evgeni) [inline bats/foreman_helper.bash:27]: You're using this in more cases than Puppet, so we either should have a more generic message here, or multiple `skip` functions with different explicit messages.

**Comment by evgeni** [comment]
the wording changes and removal of the always-skipped tests can go in a separate PR if you want, as they are not related anyways.

File: bats/foreman_helper.bash (hunk 3 above)
Comment (ehelms) [reply bats/foreman_helper.bash]: Now it supports multiple cases.

File: bats/fb-content-katello.bats [hunk 4]
```diff
@@ -245,30 +261,20 @@ setup() {
 }
 
 @test "publish content view" {
+  tSkipIfNoPulp2 "Puppet content"
```
Comment (evgeni) [inline bats/fb-content-katello.bats:264]: publish and promote should still work with pulp3 only?

File: bats/fb-content-katello.bats [hunk 5]
```diff
@@ -61,22 +61,26 @@ setup() {
   curl http://$HOSTNAME/pulp/isos/${ORGANIZATION_LABEL}/Library/custom/${PRODUCT_LABEL}/${REPOSITORY_LABEL}/1.iso > /dev/null
 }
 
-@test "create a docker repository" {
+@test "create a container repository" {
   hammer repository create --organization="${ORGANIZATION}" --docker-upstream-name="fedora/ssh" --url=https://registry-1.docker.io/ \
     --product="${PRODUCT}" --content-type="docker" --name "${DOCKER_REPOSITORY}" | grep -q "Repository created"
```
Comment (evgeni) [inline bats/fb-content-katello.bats:66]: should this then also be `CONTAINER_REPOSITORY`?

File: bats/fb-content-katello.bats (hunk 4 above)
Comment (ehelms) [reply bats/fb-content-katello.bats:264]: They should, but this publish and promote were targeted as follow ons to adding the Puppet content so they are not really testing anything new.

File: bats/foreman_helper.bash [hunk 6]
```diff
@@ -17,3 +17,13 @@ tForemanVersion() {
...
   ) | cut -d. -f1-2
 }
+
+tIsPulp2() {
+  tPackageExists pulp-server
+}
+
+tSkipIfNoPulp2() {
+  if ! tIsPulp2; then
+   skip "${1} not available with Pulp 3 only"
```
Comment (ekohl) [inline bats/foreman_helper.bash:27]:
```suggestion
   skip "${1} unavailable without Pulp 2"
```

File: bats/fb-content-katello.bats (hunk 4 above)
Comment (evgeni) [reply bats/fb-content-katello.bats:264]: oooh, right. should the test name then be adjusted to "$action content view with puppet content" or something?

**Pull request merged**